homeassistant/components/monarchmoney/entity.py

```diff
@@ -0,0 +1,79 @@
+"""Monarch money entity definition."""
+
+from typedmonarchmoney.models import MonarchAccount, MonarchCashflowSummary
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import MonarchMoneyDataUpdateCoordinator
+
+
+class MonarchMoneyEntityBase(CoordinatorEntity[MonarchMoneyDataUpdateCoordinator]):
+    """Base entity for Monarch Money with entity name attribute."""
+
+    _attr_has_entity_name = True
+
+
+class MonarchMoneyCashFlowEntity(MonarchMoneyEntityBase):
+    """Custom entity for Cashflow sensors."""
+
+    def __init__(
+        self,
+        coordinator: MonarchMoneyDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the Monarch Money Entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = (
+            f"{coordinator.subscription_id}_cashflow_{description.key}"
+        )
+        self.entity_description = description
+        self._attr_device_info = DeviceInfo(
```

Comment:
I am wondering what this device would represent, cashflow is a concept, not a service, account or device

Reply:
If you are trying to see your budget in Home Assistant you could see:

It lets you track:

- Total Earnings This Year
- Total Spending this Year
- How much I've saved
- What my Savings Rate is


So I think its kinda useful...

Reply:
```python
 data = {
        "sumIncome": 10000.0,
        "sumExpense": -5000.0,
        "savings": 5000.0,
        "savingsRate": 0.5,
    }
```

Reply:
But my as in, my user account? Or my individual bank accounts?

Reply:
It's an aggregate of all accounts in monarch that you've added. So you can see your total cash flow

Reply:
My user account

Reply:
Should it be called something tied to your account, like email?

Reply:
You can have multiple emails tied to an account the most unique thing is `subscription_id` as two+ emails can share that.

Reply:
Can we get the name of the user?

Reply:
We can use conf_email if you want... and I guess I can use unique_id to be the subscription id in the integration to stop adding dupes?

Reply:
The only way to get name is via the email it would seem -> which is probably good API design